modules/globebrowsing/src/rawtiledatareader.cpp

```diff
@@ -519,6 +522,87 @@ void RawTileDataReader::initialize() {
         }
     }
 
+    if (_cacheProperties.enabled) {
+        ZoneScopedN("MRF Caching");
+
+        std::string datasetIdentifier = std::to_string(std::hash<std::string>{}(_datasetFilePath));
+        std::stringstream path;
+        path << module.mrfCacheLocation() << "/" << _cacheProperties.path << "/" << datasetIdentifier << "/";
+        std::string root = absPath(path.str()).string();
+        std::string mrf = root + datasetIdentifier + ".mrf";
+        std::string cache = root + datasetIdentifier + ".mrfcache";
+
+        if (!std::filesystem::exists(mrf)) {
+            std::error_code ec;
+            if (!std::filesystem::create_directories(root, ec)) {
+                // Already existing directories causes a 'failure' but no error
+                if (ec) {
+                    throw ghoul::RuntimeError(fmt::format(
+                        "Failed to create directories for cache at: {}. Error Code: {}, message: {}",
+                        root, std::to_string(ec.value()), ec.message()
+                    ));
+                }
+            }
+
+            auto* driver = GetGDALDriverManager()->GetDriverByName("MRF");
```

Comment:
We refrain from using `auto` unless it is to shorten long names or when we'd otherwise duplicate the class name to increase readability;
so  -> `GDALDriver* driver = ...`